Tests/UnitTests/Identity/CustomerInfoManagerPostReceiptTests.swift

```diff
@@ -75,7 +75,57 @@ class CustomerInfoManagerPostReceiptTests: BaseCustomerInfoManagerTests {
         let parameters = try XCTUnwrap(self.mockTransactionPoster.invokedHandlePurchasedTransactionParameters.value)
 
         expect(parameters.transaction as? StoreTransaction) === transaction
-        expect(parameters.data.appUserID) == Self.userID
+        expect(parameters.data.appUserID) == Self.appUserID
+        expect(parameters.data.presentedOfferingID).to(beNil())
+        expect(parameters.data.unsyncedAttributes).to(beEmpty())
+        expect(parameters.data.source.isRestore) == false
+        expect(parameters.data.source.initiationSource) == .queue
+    }
+
+    func testReturnsLastKnownCustomerInfoIfTransactionWasAlreadyPosted() async throws {
+        let transaction = Self.createTransaction()
+
+        self.mockTransationFetcher.stubbedUnfinishedTransactions = [transaction]
+        self.mockTransactionPoster.stubbedHandlePurchasedTransactionResult.value = .alreadyPosted
+
+        self.mockDeviceCache.cachedCustomerInfo[Self.appUserID] = try self.mockCustomerInfo.jsonEncodedData
+
+        let info = try await self.customerInfoManager.fetchAndCacheCustomerInfo(appUserID: Self.appUserID,
+                                                                                isAppBackgrounded: false)
+        expect(info) == self.mockCustomerInfo
+
+        expect(self.mockBackend.invokedGetSubscriberData) == false
+        expect(self.mockTransactionPoster.invokedHandlePurchasedTransaction.value) == true
+
+        let parameters = try XCTUnwrap(self.mockTransactionPoster.invokedHandlePurchasedTransactionParameters.value)
+
+        expect(parameters.transaction as? StoreTransaction) === transaction
+        expect(parameters.data.appUserID) == Self.appUserID
+        expect(parameters.data.presentedOfferingID).to(beNil())
+        expect(parameters.data.unsyncedAttributes).to(beEmpty())
+        expect(parameters.data.source.isRestore) == false
+        expect(parameters.data.source.initiationSource) == .queue
+    }
+
+    func testFetchesCustomerInfoIfTransactionWasAlreadyPostedButNoCachedCustomerInfo() async throws {
```

Comment:
🤔 
This is a weird case to think about. Not sure fetching customer info is the best path here? 

I can only imagine arriving here if: 
- we've posted the transaction before
- we call invalidate customer info cache
- we get notified again for an unfinished transaction

Or similar thinking but someone calls `restoreCompletedTransactions`. 

Feels like it'd be safer to just post in this case? Would that be a huge refactor? WDYT?

Reply:
That complicates the code unnecessarily. I'd have to give `TransactionPoster` knowledge of whether there is a cached customer info or not.